### PR TITLE
Dedicated method allowing to add plugins to the Mesh menu

### LINKED ISSUE #46908: Missing `addPluginToMeshMenu` iface's method
### Feature description

Currently, we have to use this trick:

```python
self.mesh_menu = self.iface.mainWindow().findChild(QMenu, "mMeshMenu")
self.pluginMenu = self.mesh_menu.addMenu(QIcon(icon_path), "Menu")
self.pluginMenu.addAction(action)
```

### Additional context

Useful when creating mesh related plugin.

### PR DESCRIPTION
**Pull request by DelazJ**
Trying to fix #46908
For transparency, I just mimic what were in the other plugins menus. I hope mesh does not have any particular constraints...

### CODE REVIEW

**Comment by DelazJ** [comment]
@nicogodet Thanks. I failed to find these methods using the github search widget. I was surprised but thought there was some devs tricks behind this.
Is that normal to have all those copies of the same function (modulo menu name)? The get**xxx**menu and removePlugin**xxx**Menu  methods are afaict all the same.

File: src/gui/qgisinterface.h [hunk 1]
```diff
@@ -947,6 +947,12 @@ class GUI_EXPORT QgisInterface : public QObject
     //! Remove action from the Web menu
     virtual void removePluginWebMenu( const QString &name, QAction *action ) = 0;
 
+    //! Add action to the Mesh menu
```
Comment (m-kuhn) [inline src/gui/qgisinterface.h:950]: Can you add a `\since QGIS 3.24` here?

I think it's fine to merge this still

Comment (DelazJ) [reply src/gui/qgisinterface.h]: Done.

**Review by m-kuhn — APPROVED** [review]

File: src/app/qgisapp.cpp [hunk 2]
```diff
@@ -14721,7 +14765,7 @@ void QgisApp::addPluginToWebMenu( const QString &name, QAction *action )
   QMenu *menu = getWebMenu( name );
   menu->addAction( action );
 
-  // add the Vector menu to the menuBar if not added yet
+  // add the Web menu to the menuBar if not added yet
```
Comment (roya0045) [inline src/app/qgisapp.cpp:14768]: was this change intended? Seems out of place but I may be wrong.

Comment (DelazJ) [reply src/app/qgisapp.cpp:14768]: Not related to the main issue, but yes, intended change.
